### ISSUE TITLE
Bug in lookupActivations SQL

### ISSUE THREAD

**Issue by marty-cz** [wultra/powerauth-server#1336]
Hi,

I have noticed faulty behavior of `lookupActivations` endpoint when more than one `applicationId` in the `applicationIds` collection  is used.

The request fails with error in log:
```
ErrorResult(httpStatus=BAD_REQUEST, message=org.hibernate.hql.internal.ast.QuerySyntaxException: unexpected AST node: {vector} [SELECT a FROM io.getlime.security.powerauth.app.server.database.model.entity.ActivationRecordEntity a WHERE a.userId IN (:userIds_0) AND (:applicationIds_0, :applicationIds_1 IS NULL OR a.application.id IN (:applicationIds_0, :applicationIds_1)) AND a.timestampLastUsed < :timestampLastUsedBefore AND a.timestampLastUsed >= :timestampLastUsedAfter AND a.activationStatus IN (:states_0, :states_1, :states_2, :states_3, :states_4)]; nested exception is java.lang.IllegalArgumentException: org.hibernate.hql.internal.ast.QuerySyntaxException: unexpected AST node: {vector} [SELECT a FROM io.getlime.security.powerauth.app.server.database.model.entity.ActivationRecordEntity a WHERE a.userId IN (:userIds_0) AND (:applicationIds_0, :applicationIds_1 IS NULL OR a.application.id IN (:applicationIds_0, :applicationIds_1)) AND a.timestampLastUsed < :timestampLastUsedBefore AND a.timestampLastUsed >= :timestampLastUsedAfter AND a.activationStatus IN (:states_0, :states_1, :states_2, :states_3, :states_4)])
```
The problematic part of the SQL is
```
... AND (:applicationIds_0, :applicationIds_1 IS NULL OR a.application.id IN (:applicationIds_0, :applicationIds_1)) ...

---> :applicationIds_0, :applicationIds_1 IS NULL
```
Location of problem in code: https://github.com/wultra/powerauth-server/blob/1.6.1/powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/ActivationRepository.java#L175

May I ask you for a fix or a workaround solution? 
Thanks

**Comment by zcgandcomp** [comment]
Please put to the tests multiple `applicationId` scenario.

Plus, a backport is needed.

**Comment by marty-cz** [comment]
I have just spotted we are using version 1.1.1 and not the latest one. So the issue is probably fixed and we need to upgrade to latest version.

**Comment by jandusil** [comment]
I was able to reproduce this issue in v1.1.x. The problem lies in the missing parentheses around `:applicationIds` in the query. This issue was fixed by commit 8108d4d28caa971f0fde90505a4c82cf3e82df31, which is part of v1.3.x. Therefore, upgrading to this version should resolve the problem.

**Comment by jandusil** [comment]
I was able to reproduce this issue in v1.1.x. The problem lies in the missing parentheses around `:applicationIds` in the query. This issue was fixed by commit 8108d4d28caa971f0fde90505a4c82cf3e82df31, which is part of v1.3.x. Therefore, upgrading to this version should resolve the problem.
@marty-cz 
CC: @zcgandcomp
